### DISCUSSION TITLE
rust

### DISCUSSION THREAD

**Discussion by ScarsatoAndrea** [drknzz/GitHub-Achievements#319] [Q&A] [answered]
Is rust good for game dev?

**Comment by SimVac** [comment, marked as answer]
i don't know, i have never used it for game dev

**Reply by ScarsatoAndrea** [reply]
thx

**Comment by AsherBond** [comment]
Some games written in Rust:

Star Citizen: An ambitious space simulation game with a massive open world, developed by Cloud Imperium Games.
Kingdom Come: Deliverance: A historical RPG set in medieval Bohemia, developed by Warhorse Studios.
Satisfactory: A first-person open-world factory building game, developed by Coffee Stain Studios.
Factorio: A strategy game where players build and maintain factories, developed by Wube Software.
Prison Architect: A prison management simulation game, developed by Introversion Software.
Kerbal Space Program: A space flight simulator game, developed by Squad.
Cities: Skylines: A city-building simulation game, developed by Colossal Order.
Planet Coaster: A theme park simulation game, developed by Frontier Developments.
Jurassic World Evolution: A dinosaur park management simulation game, developed by Frontier Developments.
Advantages of Rust:

Memory safety: Rust's ownership system and borrow checker eliminate entire classes of memory-related bugs, making games more reliable and secure.
Performance: Rust's compiled nature and zero-cost abstractions allow for performance comparable to C and C++, crucial for demanding game engines.
Concurrency: Rust's concurrency features enable efficient multi-threading, essential for handling complex game logic and rendering tasks.
Modern tooling: Rust's modern compiler, package manager, and build system streamline development and improve developer productivity.
Community: Rust's active and supportive community provides valuable resources and assistance to game developers.

Disadvantages of Rust:

Steep learning curve: Rust's ownership system and borrow checker can be challenging for new developers to grasp, requiring more time to learn.
Compiler: Rust's compile times can be longer than some other languages, especially for large game projects. More importantly what it takes to get something to compile requires more effort; however, this isn't really a disadvantage in my opinion. I'm putting it here because I feel like I'm taking a contrarian stance, but honestly the time you spend getting stuff to compile will save you time troubleshooting the running code later.
Limited ecosystem: It's really the biggest disadvantage and may go away over time. While Rust's game development ecosystem is growing, it's still smaller compared to more established languages, with fewer game-specific libraries and tools available. For example if you were using godot or something more geared toward game development rather than runtime resilience, memory safety, and reliability you could hit the ground running faster but you might keep hitting the ground if you have too many errors. With easier game dev frameworks you might be jumpin' out the gym like Jacky Joyner, but you're likely also dereferencing a null pointer.
